client/app/telecom/telephony/line/phone/programmableKeys/telecom-telephony-line-phone-programmableKeys.controller.js

```diff
@@ -85,7 +85,16 @@ export default class TelecomTelephonyLinePhoneProgammableKeysCtrl {
     return this.line.getPhone().then(() => {
       if (this.line.hasPhone) {
         return this.line.phone.initDeffered().then(() => {
-          this.functionKeys.raw = angular.copy(this.line.phone.functionKeys);
+          this.functionKeys.raw = _.cloneDeep(this.line.phone.functionKeys);
+          this.functionKeys.raw.sort((a, b) => {
+            const regexp = new RegExp(/[a-z]+|\d+/ig);
```

Comment:
small trick but if you declare your regex outside of the anonymous function it would be faster by avoiding to instantiate a new regex on each comparison